### PR TITLE
MSP-3578: add ssh to slurm work

### PR DESCRIPTION
**Pull request by Uburro**

### CODE REVIEW

File: api/v1/slurmcluster_types.go [hunk 1]
```diff
@@ -635,6 +639,18 @@ type SlurmNodeWorker struct {
 	CgroupVersion string `json:"cgroupVersion,omitempty"`
 }
 
+type SlurmNodeWorkerSshd struct {
+	// Enabled defines whether the SSH sidecar is enabled on the worker node
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
+	// Sshd represents the SSH daemon service configuration
```
Comment (asteny) [inline api/v1/slurmcluster_types.go]: There is no Sshd configuration here...

File: internal/render/common/volume.go [hunk 2]
```diff
@@ -300,3 +300,120 @@ func RenderVolumeMountRESTJWTKey() corev1.VolumeMount {
 }
 
 // endregion REST JWT key
+
+// region SSH
+
+// region configs
+
+// renderVolumeSshConfigs renders [corev1.Volume] containing SSH configs contents
```
Comment (asteny) [inline internal/render/common/volume.go:308]: sshd

File: internal/render/common/volume.go [hunk 3]
```diff
@@ -300,3 +300,120 @@ func RenderVolumeMountRESTJWTKey() corev1.VolumeMount {
...
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: naming.BuildConfigMapSSHConfigsName(clusterName),
+				},
+			},
+		},
+	}
+}
+
+// renderVolumeMountSshConfigs renders [corev1.VolumeMount] defining the mounting path for SSH configs
```
Comment (asteny) [inline internal/render/common/volume.go]: sshd

File: internal/render/common/volume.go [hunk 4]
```diff
@@ -300,3 +300,120 @@ func RenderVolumeMountRESTJWTKey() corev1.VolumeMount {
...
+		MountPath: consts.VolumeMountPathSSHConfigs,
+		ReadOnly:  true,
+	}
+}
+
+// endregion configs
+
+// region root keys
+
+// renderVolumeSshRootKeys renders [corev1.Volume] containing SSH root keys contents
```
Comment (asteny) [inline internal/render/common/volume.go]: sshd

File: internal/render/common/volume.go [hunk 5]
```diff
@@ -300,3 +300,120 @@ func RenderVolumeMountRESTJWTKey() corev1.VolumeMount {
...
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: naming.BuildConfigMapSshRootPublicKeysName(clusterName),
+				},
+				DefaultMode: ptr.To(int32(384)),
+			},
+		},
+	}
+}
+
+// renderVolumeMountSshRootKeys renders [corev1.VolumeMount] defining the mounting path for SSH root keys
```
Comment (asteny) [inline internal/render/common/volume.go]: sshd

File: internal/render/worker/container.go [hunk 6]
```diff
@@ -174,3 +175,57 @@ func renderRealMemorySlurmd(resources corev1.ResourceRequirements) int64 {
 	memoryInMebibytes := memoryInBytes / 1_048_576 // 1 MiB = 1,048,576 bytes
 	return memoryInMebibytes
 }
+
+func renderContainerSshd(
```
Comment (asteny) [inline internal/render/worker/container.go]: I think that this part same as in login node... Can we use it one time from common?

**Review by asteny — CHANGES_REQUESTED** [review]

File: internal/render/worker/container.go (hunk 6 above)
Comment (Uburro) [reply internal/render/worker/container.go]: they have different arguments.  i do not think we have to spend  time for this way

**Pull request closed without merging**